dml_events.go

```diff
@@ -423,6 +423,11 @@ func appendEscapedString(buffer []byte, value string) []byte {
 
 func appendEscapedBuffer(buffer, value []byte, isJSON bool) []byte {
 	if isJSON {
+		// TODO: link to issue
+		if len(value) == 0 {
+			value = []byte("null")
```

Comment:
This would be changing the actual data on MySQL from the "invalid null value" that it stores to a real `'null'` value. I think this is fine because half of MySQL already treats the "invalid null value" as `'null'` and one of the only place this is visible is within the replication.

Reply:
> However, MySQL will silently change this to a JSON-type null value instead of an actual NULL or empty value.

Not quite. MySQL will internally store the empty string and when will transform it to the JSON `'null'` when serving the value to the consumer. It just didn't do this conversion in the binlog in 5.7.

Reply:
Yes, I omitted the `on read` part from the end. Thanks!